lib/functionsEmulator.js

```diff
@@ -0,0 +1,191 @@
+'use strict';
```

Comment:
The main things to note are:
- URL for HTTPS functions only gets printed out for firebase serve, not for firebase functions:shell
- firebase serve only serves HTTPS functions, firebase functions:shell emulates all functions